datadog-integrations-aws-handler/src/datadog_integrations_aws/handlers.py

```diff
@@ -73,6 +74,16 @@ def create_handler(
                 message=f"Error creating AWS account: {e}",
                 errorCode=http_to_handler_error_code(e.status)
             )
+    if model.ExternalIDSecretName is not None:
+        secret_name = model.ExternalIDSecretName
+    else:
+        secret_name = DEFAULT_SECRET_NAME
+    boto_client = session.client("secretsmanager")
```

Comment:
The session's client already has the correct region associated with it: https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/master/src/cloudformation_cli_python_lib/boto3_proxy.py